src/cli.js

```diff
@@ -1,103 +1,196 @@
-    .alias('h', 'help').help('h')
-    .version(() => require('../package.json').version)
-let argv = yarg.argv;
-let creator;
-
-function writeFile(f) {
-    getSource(f).then((content) => {
-        creator.create(f, content, !!argv.w)
-        .then(content => content.writeFile())
-        .then(content => {
-            console.log('Wrote ' + chalk.green(content.outputFilePath));
-            content.messageList.forEach(message => {
-                console.warn(chalk.yellow('[Warn] ' + message));
-            });
-        })
-        .catch(reason => console.error(chalk.red('[Error] ' + reason)));
-    });
-};
+const readSass = (file, relativeTo) => (
+    new Promise((resolve, reject) => {
+        sass.render(
+            { file },
+            (err, result) => {
+                if (err && (relativeTo && relativeTo !== '/')) {
+                    return resolve([]);
+                } else if (err && (!relativeTo || relativeTo === '/')) {
+                    return reject(err);
+                }
+                return resolve(result.css.toString());
+            },
+        );
+    })
+);
 
-function getSource(file, relativeTo) {
-    return new Promise((resolve, reject) => {
-      sass.render({file}, (err, result) => {
-        if (err && relativeTo && relativeTo !== '/') {
-          return resolve([]);
+const createTypings = (f, creator, cache, handleError, handleWarning, verbose) => (
+    readSass(f)
+    .catch((reason) => {
+        handleError(`${chalk.red(f)}\n${reason}\n`);
+    })
+    .then(content => creator.create(f, content, cache))
+    .then(c => c.writeFile())
+    .then((c) => {
+        if (verbose) {
+            console.info(`Created ${chalk.green(c.outputFilePath)}`);
         }
-  
-        if (err && (!relativeTo || relativeTo === '/')) {
-          return reject(err);
+        c.messageList.forEach((message) => {
+            handleWarning(`${chalk.yellow(f)}\nWarning: ${message}\n`);
+        });
+        return c;
+    })
+    .catch((reason) => {
+        handleError(`${chalk.red(f)}\nError: ${reason}\n`);
+    })
+);
+
+
+const createTypingsForFileOnWatch = (creator, cache, verbose) => (f) => {
+    let warnings = [];
+    let errors = [];
+
+    const cleanUp = () => {
+        warnings.forEach(m => console.warn(m));
+        errors.forEach(e => console.error(e));
+        errors = [];
+        warnings = [];
+    };
+
+    const handleError = (e) => { errors.push(e); };
+    const handleWarning = (w) => { warnings.push(w); };
+    return createTypings(f, creator, cache, handleError, handleWarning, verbose)
+        .then(cleanUp);
+};
+
+const createTypingsForFiles = (creator, cache, verbose) => (files) => {
+    let errors = [];
+    let warnings = [];
+
+    const cleanUp = () => {
+        warnings.forEach(m => console.warn(m));
+        errors.forEach(e => console.error(e));
+        if (warnings.length + errors.length > 0) {
```

Comment:
not the right place to output warnings and errors. To keep track of the total number of messages, you can use a counter instead. I also wouldn't call it cleanUp, it's more like making a summary message and you clean up after it.